transmart-core-db/grails-app/services/org/transmartproject/db/clinical/MultidimensionalDataResourceService.groovy

```diff
@@ -681,13 +418,18 @@ class MultidimensionalDataResourceService implements MultiDimensionalDataResourc
                 .insertResultToTable(QtPatientSetCollection, ['patient.id', 'resultInstance.id'], patientSetCriteria)
     }
 
-    @Override QueryResult findQueryResult(Long queryResultId, User user) {
-        QueryResult queryResult = QtQueryResultInstance.findById(queryResultId)
+    @Override
+    QueryResult findQueryResult(Long queryResultId, User user) {
```

Comment:
Now, actually, it returns something else. `findFinishedPatientSetQueryResult(...)`
It change contract of `GET /v2/patient_sets/${id}` call as this method is reused.

Reply:
Which is intentional. Actually `GET /v2/patient_sets` should also only return finished patient sets.

Reply:
Yes, actually, they are always finished currently. As the call is synchronous.

Reply:
Unless multiple concurrent requests of course.

Reply:
And then you only want the finished ones.